pkg/booleanpolicy/default_policies_test.go

```diff
@@ -3208,7 +3208,13 @@ func (suite *DefaultPoliciesTestSuite) TestNetworkPolicyFields() {
 			v2 := suite.getViolations(missingEgressPolicy, enhanced)
 
 			allAlerts := append(v1.AlertViolations, v2.AlertViolations...)
-			suite.Equal(testCase.alerts, allAlerts)
+			for i, expected := range testCase.alerts {
+				suite.Equal(expected.GetType(), allAlerts[i].Type)
+				suite.Equal(expected.GetMessage(), allAlerts[i].Message)
+				suite.Equal(expected.GetKeyValueAttrs(), allAlerts[i].GetKeyValueAttrs())
```

Comment:
I changed the assertion from `DeepEqual` to `Equal` on selected fields to avoid comparing alert time - the timestamp is generated dynamically and I do not want to assert on the value - I just want that the value exists.

So even having the `GetKeyValueAttrs()` always returning nil, I think that the assertion may make sense to mimick the original DeepEqual as closely as possible.

Reply:
But let me maybe add some new test cases where `GetKeyValueAttrs` would not be nil, so that we cover that part too.

Reply:
I added one case in [713e2ac](https://github.com/stackrox/stackrox/pull/1219/commits/713e2acad678f75e07059b8af1d2f2649e49f1b8) - let me know if that is okay

Reply:
Looks good